util/runpytest.sh

```diff
@@ -0,0 +1,173 @@
+
+# This environment variable (should) make the driver test not try to use real hardware (only simulator)
+export TEST_NOHW=1
+
+# This environment variable makes the bugreporter test skip test cases that involve ticket creation
+export TEST_NO_SUPPORT_TICKET=1
+
+# A random number, which is always the same, in order to force the dict order in Python 3.
+# This helps in reproducing issues, by making each run a little less different.
+# Note that this doesn't affect the back-end (as it's run in a separate user),
+# and there are many other things that can affect execution order (eg, threads).
+export PYTHONHASHSEED=1567315
+
+mkdir -p ~/development/odemis-testing
+TESTLOG=~/development/odemis-testing/unittest-full-$DATE.log
+TESTSUMMARY=~/development/odemis-testing/pytest-summary-$DATE.log
+# make sure it is full path
+TESTLOG="$(readlink -m "$TESTLOG")"
+TESTSUMMARY="$(readlink -m "$TESTSUMMARY")"
+
+
+# The temporary files are used to save multi line strings (This had issues with bash)
+rm /tmp/filtered_test.txt /tmp/test_summary.txt
+touch /tmp/test_summary.txt /tmp/filtered_test.txt
+
+
+if [ ! -d /var/run/odemisd ] ; then
+    echo  "Need /var/run/odemisd"
+    sudo mkdir -m 777 /var/run/odemisd
+fi
+
+# stop the backend
+sudo odemis-stop
+
+# find the test scripts (should not contain spaces)
+testfiles="$(find "$ODEMIS_SRC" -wholename "*/test/*test.py")"
+
+#Warn if some files are misnamed
+skippedfiles="$(find "$ODEMIS_SRC" -wholename "*/test/*.py" -and -not -wholename "*/test/*test.py")"
+if [ "$skippedfiles" != "" ]; then
+    echo "Warning, these scripts are not named *_test.py and will be skipped:"
+    echo "$skippedfiles"
+fi
+
+run_unittests()
+{
+interpreter=$1
+
+echo -e "\n\n==============================================="
+echo "Running unit tests in $interpreter"
+echo "Running unit tests on $(date)" > "$TESTLOG"
+
+# run each test script and save the output
+failures=0
+for f in $testfiles; do
+    echo "Running $f..."
+    if ! grep -q "__main__" $f; then
+        echo "WARNING: test $f seems to not be runnable"
+    fi
+    echo "Running $f:" >> "$TESTLOG"
+    prev_size=$(wc -l < "$TESTLOG")
+    # run it in its own directory (sometimes they need specific files from there)
+    pushd "$(dirname $f)" > /dev/null
+        # Automatically kill after MAXTIME, then try harder after 30 s
+        timeout -k 30 $MAXTIME $interpreter -m pytest $f --tb=line --verbose >> "$TESTLOG" 2>&1
+        status=$?
+        echo $f returned $status >> "$TESTLOG" 2>&1
+    popd > /dev/null
+    # Don't show test output if the file hasn't grown, as it'd be the previous test output
+    new_size=$(wc -l < "$TESTLOG")
+    if [[ "$new_size" == "$prev_size" ]]; then
+        echo "NOT RUN"
+    else
+        tail -n "+$prev_size" "$TESTLOG" > /tmp/latest_test_case_log.txt
+        # Filter the output to only print the summary
+        python3 ~/development/pytest/log_filter.py /tmp/latest_test_case_log.txt > /tmp/filtered_test.txt
```

Comment:
Is `~/development/pytest/log_filter.py` the same as `~/development/odemis/util/pytest_log_filter.py`?

Reply:
Nice catch. This only worked because I created a folder before adding it to odemis.

Reply:
Don't hardcode the path. You can use `$ODEMIS_DIR/util/...`